### ISSUE TITLE
[Question] Can I change the transparency of all handwriting uniformly?

### ISSUE THREAD

**Issue by iOSyan** [hm21/pro_image_editor#55]
Can I change the transparency of all handwriting uniformly? 
Also, when I save the currently edited image, can I modify the previous edits the next time I open it?
thx!

**Comment by hm21** [comment]
Yes, you can export the state history and import that state history the next time, allowing you to change the previous edits. The example in the dev branch [here](https://github.com/hm21/pro_image_editor/blob/dev/example/lib/pages/import_export_example.dart) shows you how to do it.

**Comment by iOSyan** [comment]
Thanks for your reply and thanks for the wonderful project，it's help a lot…

Can I change the transparency of all handwriting uniformly?

**Comment by hm21** [comment]
Currently, it's not possible to change the transparency from the drawing. However, I have put it on my roadmap [here](https://github.com/users/hm21/projects/5/views/1?filterQuery=) and will release it with the new version `3.0.0` which allows you to set the opacity.

**Comment by iOSyan** [comment]
Okay, thank you very much. My current project requirement is to adjust the transparency of all handwriting, including those that have already been drawn. Do I currently have any methods to implement this feature?

**Comment by hm21** [comment]
Currently, there's no way to do this. But I can try to add this feature for you. 
If you have multiple drawings, do you need to set a different opacity for each drawing or the same opacity for all?

**Comment by iOSyan** [comment]
You are so kind, thank you so much!
I just reviewed the project requirements again and found that this requirement is not currently required. Sorry~~~~

**Comment by iOSyan** [comment]
Sorry to bother you again.
Is it possible to listen to the text input by the user when using the text function? (The project needs to process the text input by the user). For example, covertly converting the user's input of '#1' to blue. (the user cannot notice this change through the app).

Thank you very much.

**Comment by hm21** [comment]
Currently, it's not possible to listen to the text input field. However, I will add a callback for you where you can listen to it. I can't say exactly when I'll have time to do it, but I'll definitely do it today and text you here after I upload it. I will also post then an example of how you can listen to it.

**Comment by hm21** [comment]
Okay, I have now released version `3.0.0-dev.17` where you can listen to a callback like the one below.

```dart
ProImageEditor.asset(
      'demo.jpg',
      key: editorKey,
      callbacks: ProImageEditorCallbacks(
        textEditorCallbacks: TextEditorCallbacks(
          onChanged: (value) {
            print(value);
          },
          onEditingComplete: () {
            print('onEditingComplete');
          },
          onSubmitted: (value) {
            print(value);
          },
        ),
    ),
)
```
 
Btw, if you have any further questions, I'd appreciate it if you could initiate a new Q&A thread in the [discuss-tab](https://github.com/hm21/pro_image_editor/discussions). This would be beneficial for other users, allowing them to find answers to similar questions directly.

**Comment by iOSyan** [comment]
Thank you so much!!
And if I have any further questions, I will initiate a new Q&A thread in the [discuss-tab](https://github.com/hm21/pro_image_editor/discussions).
